.github/workflows/puppetfile.yml

```diff
@@ -0,0 +1,73 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Puppetfile checks
+
+# Controls when the workflow will run
+on:
+  workflow_call:
+    inputs:
+      container_image:
+        description: Image to use when running tests
+        default: 'puppet/puppet-dev-tools:2022-03-28-92c7176'
+        required: false
+        type: string
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  r10k_validate:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    container: ${{ inputs.container_image }}
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+#      - name: github known host
+#        run: mkdir /~/.ssh/ && echo githubPubKey >> /~/.ssh/known_hosts
+      - name: validate Puppetfile
+        run: rake -f /Rakefile r10k:validate
+  r10k_check_dups:
```

Comment:
Is there are reason you're not using multiple steps but a new job for every step? This means starting a new container every time + a new checkout, which is slower. You could even run `rake -f /Rakefile r10k:validate r10k:duplicates r10k:syntax r10k:deprecations` in a single job.

Reply:
Steps do not run in parallel and some of the rake tasks take some time to complete so it could end up taking longer to combine all of them.  Using jobs,  runs all the rake tasks in parallel which is faster if the jobs take longer.  Additionally, each job can be restarted independently.

Without breaking out the jobs I would expect the overall time to either be the same or longer.  Although this really depends on the number of jobs.

It does start a new container and when run on unique runners, pulls the container image for every job (not ideal).   Which at least happens in parallel. If a custom runner was supplied this container image would be cached.